### PR TITLE
Stories/dt/fixkeywordregex 108

### LINKED ISSUE #108: filter \ in keywordregex
currently the function to create the keyword regex can break if a \ is used in the keyword. 
Replace this (and any other characters that can break regex) to prevent breaking the regex.

### PR DESCRIPTION
**Pull request by terw-dan**
Fixes #108. 
Adds filter to the regexes to filter out unwanted characters (- [ ] / { } ( ) * + ? . \^ $ |) that could break the regex, or return unexpected results since these are keycharacters in the regex.

### CODE REVIEW

File: dist/yoast-seo.js [hunk 1]
```diff
@@ -2182,13 +2186,17 @@ YoastSEO.SnippetPreview.prototype.getPeriodMatches = function() {
...
 /**
  * formats the keyword for use in the snippetPreview by adding <strong>-tags
+ * strips unwanted characters that could break the regex or give unwanted results
  * @param textString
  * @returns textString
  */
 YoastSEO.SnippetPreview.prototype.formatKeyword = function( textString ) {
 
+	// removes characters from the keyword that could break the regex, or give unwanted results.
+	var keyword = this.refObj.rawData.keyword.replace( /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "" );
```
Comment (CarolineGeven) [inline dist/yoast-seo.js]: Is it possible to save this in a class variable so you don't need to set the regex everytime? In this file alone, this exact regex occurs 3 times. If you need to change the regex, you have to change it in 3 spots as well. It'd be better to have 1 place where this is set and changed.

**Comment by CarolineGeven** [comment]
CR1 done

**Comment by CarolineGeven** [comment]
CR2 done :+1:

**Comment by diedexx** [comment]
Acceptance Done. @CarolineGeven Could you please check and confirm?

File: dist/yoast-seo.js [hunk 2]
```diff
@@ -2481,6 +2492,11 @@ YoastSEO.StringHelper.prototype.stripAllTags = function( textString ) {
 	return textString;
 };
 
+YoastSEO.StringHelper.prototype.cleanRegex = function( textString ) {
```
Comment (omarreiss) [inline dist/yoast-seo.js]: I don't know which regex this cleans, but to me it seems like this function sanitizes a string. @atimmer please rename accordingly.

File: dist/yoast-seo.js [hunk 3]
```diff
@@ -2197,12 +2206,14 @@ YoastSEO.SnippetPreview.prototype.formatKeyword = function( textString ) {
...
  * <strong>-tags
+ * strips unwanted characters that could break the regex or give unwanted results
  *
  * @param textString
  * @returns {XML|string|void}
  */
 YoastSEO.SnippetPreview.prototype.formatKeywordUrl = function( textString ) {
-	var replacer = this.refObj.rawData.keyword.replace( " ", "[-_]" );
+	var keyword = this.refObj.stringHelper.cleanRegex( this.refObj.rawData.keyword );
+	var replacer = keyword.replace( " ", "[-_]" );
```
Comment (omarreiss) [inline dist/yoast-seo.js]: What is a replacer? This is a keyword, and it's sanitized so maybe something like `sanitizedKeyword` will do. @atimmer please rename accordingly.

File: dist/yoast-seo.js [hunk 4]
```diff
@@ -2182,13 +2187,17 @@ YoastSEO.SnippetPreview.prototype.getPeriodMatches = function() {
...
  * @returns textString
  */
 YoastSEO.SnippetPreview.prototype.formatKeyword = function( textString ) {
 
+	// removes characters from the keyword that could break the regex, or give unwanted results, includes the -
+	var keyword = this.refObj.rawData.keyword.replace( /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "" );
+
 	//matches case insensitive and global
-	var replacer = new RegExp( "\\b" + this.refObj.rawData.keyword + "\\b", "ig" );
+	var replacer = new RegExp( "\\b" + keyword + "\\b", "ig" );
```
Comment (omarreiss) [inline dist/yoast-seo.js]: Isn't this a regex?

**Comment by omarreiss** [comment]
@atimmer please fix comments and merge.

**Pull request merged**